bin/ruby-build

```diff
@@ -1285,8 +1285,8 @@ build_package_enable_yjit() {
 
   local rustc_ver="$(rustc --version 2>/dev/null)"
   [ -n "$rustc_ver" ] || return 0
-  # Some kind of built-in bash comparison for dotted version strings would be awesome.
-  if [[ "${rustc_ver}" == *"rustc 1."[6789]* ]]; then
+  # YJIT supports Rust 1.58.1 and later
+  if [[ "${rustc_ver}" =~ ^(rustc 1\.)(58\.1|59*|[6789]*) ]]; then
```

Comment:
It seems like `*` is not glob but a normal `*` of regex.
```
$ [[ "rustc 1.58.0 (123abcdef 2022-08-31)" =~ ^(rustc 1\.)(58\.1|59*|[6789]*) ]]; echo $?
0
$ [[ "rustc 1.58.1 (123abcdef 2022-08-31)" =~ ^(rustc 1\.)(58\.1|59*|[6789]*) ]]; echo $?
0
```

It should probably be something like:

```suggestion
  if [[ "${rustc_ver}" =~ ^rustc\ 1\.(58\.1|59\.|[6789]\d\.) ]]; then
```

(Ignoring >= 2 and >= 1.100 for now)